### DISCUSSION TITLE
Which tests for which roles in certification?

### DISCUSSION THREAD

**Discussion by EricSTMicro** [openthread/openthread#7961] [Q&A] [answered]
Hello,

I have 3 different binaries: FTD, MTD, RCP.

I would like to pass the Thread certification with GRL, but I'm a bit lost on which test are done for each role by the test house.

When applying for the certification, we choose what category to apply for between these one:
   - Minimal Thread Device – MED (Minimal End Device) role
   - Minimal Thread Device – SED (Sleepy End Device) role
   - Minimal Thread Device – SSED (Synchronous Sleepy End Device) role
   - Full Thread Device [REED behavior] - Leader, Router, REED roles
   - Full Thread Device [FED behavior] – FED (Full End Device) role
   - Border Router
   - On-Mesh Commissioner
   - Thread Joiner

But we don't know exactly which tests are done exactly. For example, where are tested the end-device role? For a Border Router, do we also need to pass the FTD tests?

Here are the roles I intend to pass in my internal pre-certification, which I want to match the official certification done by the test house:
- **FTD:**
   - OT1.1: Leader, router, REED, commissioner, joiner, end-device, end-device FED, end-device MED
   - OT1.2: Leader, router, REED, joiner, end-device FED, end-device MED

- **MTD:**
   - OT1.1: Joiner, end-device, end-device MED, SED
   - OT1.2: Joiner, end-device MED, SED, SSED

- **RCP:**
   - OT1.1: Border router, + FTD tests (?)
   - OT1.2: BR_1, BR_2, + FTD tests (?)

Is this right?

If I'm confused is it because I get contradictory information from different official documents.
A clue on which one is the "true" one?

BB,
Eric

**Comment by jwhui** [comment, marked as answer]
I would reach out to Tom Sciorilli, Thread Group's Director of Certification, for definitive answers to these questions.

A Border Router typically operates as an FTD, so those tests should be in scope.

A FTD may or may not operates as an MED/SED, so that depends on what roles you are looking to certify.

**Reply by EricSTMicro** [reply]
Thank you for your quick answer.
I'll try to get an answer from him then.

Eric
